### PR TITLE
Adopt PEP517 and PEP621 

### PR DESCRIPTION
**Pull request by Zeitsperre**
## Overview

Changes:

* Added a `pyproject.toml` configuration to replace `setup.py`
* Converted `CONTRIBUTING.md` to ReStructuredText and placed at top-level
* Added a few package goodies to the documentation build configuration
* Removed Travis CI configuration and obsolete `MANIFEST.in`
* Updated the `Makefile` to no longer use the `with-binaries` RavenPy hack
* Dropped Cruft

## Related Issue / Discussion

The amount of changes currently in this PR are monumental, and the `cruft` configuration is no longer relevant as a result. I can re-add this back in, but the source recipe should be fully updated to support `pyproject.toml` and other configurations.

## Additional Information

https://peps.python.org/pep-0517/
https://peps.python.org/pep-0621/

### CODE REVIEW

**Review by richardarsenault — COMMENTED** [review]
I think I understand most of these but I am unsure for some elements. I'd rather let someone more experienced give their OK!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -2,81 +2,81 @@ default_language_version:
...
-    -   id: flake8
+      - id: flake8
         args: [ '--config=setup.cfg' ]
--   repo: https://github.com/psf/black
+  - repo: https://github.com/psf/black
     rev: 23.3.0
     hooks:
-    -   id: black
+      - id: black
         args: [ '--target-version=py38' ]
```
Comment (huard) [inline .pre-commit-config.yaml:42]: should this use 3.9 as well ?

**Review by huard — APPROVED** [review]
Very nice !

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Zeitsperre) [reply .pre-commit-config.yaml:42]: Indeed! Will address.

**Review by tlvu — APPROVED** [review]
LGTM.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -36,7 +36,10 @@ jobs:
         shell: bash -l {0}
     strategy:
       matrix:
-        python-version: [ "3.8", "3.9", "3.10" ]
+        python-version:
+         - "3.9"
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply .github/workflows/main.yml:40]: Ok will do.

[the remark this answers is not in the document]
Comment (huard) [reply .github/workflows/main.yml]: uh ho, ravenpy will fail on 3.8 as of now. It's nothing big, but some of the typing annotations that Trevor recently introduced are not 3.8 compatible.

[the remark this answers is not in the document]
Comment (Zeitsperre) [reply .github/workflows/main.yml]: Really? Crud.

[the remark this answers is not in the document]
Comment (Zeitsperre) [reply .github/workflows/main.yml]: My changes were made with the impression that we were effectively dropping Python3.8 in RavenPy. Let me see where the problems are in RavenPy and I can make some backwards-compatible changes (I don't imagine there are many).

[the remark this answers is not in the document]
Comment (huard) [reply .github/workflows/main.yml]: Thanks !

**Pull request merged**